### DISCUSSION TITLE
paging-demo-upgraded-to-aspnet-mvc-rc/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#27] [Blog comments]
# paging-demo-upgraded-to-aspnet-mvc-rc/

https://blogs.taiga.nl/martijn/posts/paging-demo-upgraded-to-aspnet-mvc-rc/

**Comment by disqus-comments-migrator** [comment]
<em>Edward Johnson commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 13 2009, 02:17</em>

---

<p>When I click on the button for a page it renders the page but the highlighted (therefore current) page is always one higher than the one I clicked on.</p><p>Haven't traced it down but it seems that if I'm on page 2 and click on page 1 the logic is looking at the current page (currently 2), not the page requested</p>

**Comment by disqus-comments-migrator** [comment]
<em>Antonio Tirado commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 17 2009, 12:57</em>

---

<p>Hi, I'm looking at your code and it seems that if you're using an actual database you'd be querying it each time you requested a page. Is this so? How do would you get around this constant hammering using MVC?</p><p>Thanks in advance!</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 17 2009, 01:57</em>

---

<p>@Antonio: yes, you'd be querying the database each time, but you have to make sure that the database only returns the data for a specific page.<br>With Linq2SQL this happens automatically because the pager supports IQueryable. When using NHibernate or some other custom data access  method, you have to query for the data for the specific page and also query for the total number results (totalCount). The totalCount value can be passed into the PagedList constructor, together with the data of the specific page.</p>

**Comment by disqus-comments-migrator** [comment]
<em>Robert commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at maart 29 2009, 01:49</em>

---

<p>There seems to be a bug in paginator related to the fact that it does not shows proper "..." if, for example, are 122 items in datasource. Corresponding number of pages = 13.</p><p>If current page is 1, the paginator will display 1,2,3,4,5,6,7,8,9,10 instead of 1,2,3,4,5,6,7,8...12,13.</p><p>The same when current page is last page, the paginator displays 3,4,5,6,7,8,9,10,11,12,13 instead of 1,2...,6,7,8,9,10,11,12,13.</p>
